torch/autograd/_functions/reduce.py

```diff
@@ -3,78 +3,75 @@
 from ..function import Function
 
 
-class _DimReduceFunction(Function):
-
-    def __init__(self, dim=None):
-        super(_DimReduceFunction, self).__init__()
-        self.dim = dim
-
-    def forward(self, input):
-        self.input_size = input.size()
-        fn = getattr(input, self.fn_name)
-        if self.dim is None:
-            return input.new((fn(),))
+class Sum(Function):
+
+    @staticmethod
+    def forward(ctx, input, dim=None):
+        ctx.dim = dim
+        ctx.input_size = input.size()
+        if dim is None:
+            return input.new((input.sum(),))
         else:
-            return fn(self.dim)
-
-
-class Sum(_DimReduceFunction):
-    fn_name = 'sum'
+            return input.sum(dim)
 
-    def backward(self, grad_output):
-        if self.dim is None:
-            return grad_output.new(self.input_size).fill_(grad_output[0])
+    @staticmethod
+    def backward(ctx, grad_output):
+        if ctx.dim is None:
+            return grad_output[0].expand(ctx.input_size), None
```

Comment:
@apaszke you mentioned that this can break the graph. can you tell me how again?